app/controllers/carts_controller.rb

```diff
@@ -1,49 +1,103 @@
 class CartsController < ApplicationController
-  before_action :session_cart, :load_product,
-                only: %i(add_to_cart remove_from_cart)
-  before_action :cart_include_product?, only: %i(remove_from_cart)
+  before_action :load_product,
+                only: %i(add_to_cart delete_item_cart minus_quantity_cart)
+  before_action :sold_out?, :max_quantity, only: %i(add_to_cart)
+  before_action :session_cart, only: %i(add_to_cart)
+  before_action :cart_include_product?,
+                only: %i(delete_item_cart minus_quantity_cart)
+
+  def index
+    store_location
+  end
 
   def add_to_cart
     id = params[:id]
     session_cart.include?(id) ? add_exist_product(id) : add_new_product(id)
   end
 
-  def remove_from_cart
+  def minus_quantity_cart
```

Comment:
check lại nhé, thử truyền vào 1 id product ko tồn tại trong DB xem nó có xử lý đúng cho action này ko bé nhé

tìm hiểu về `request.xhr?` nhé
suggest cú pháp
`render(:js => "window.location = '#{name_path}'") if request.xhr?`

Reply:
dạ em có 1 cái before_action cho action này là load_product rồi í chị, không có trong db thì nó đá về trang home và toastr message không tìm thấy í chị

Reply:
em hiểu nhầm í chị rồi =))
em có test rồi, nhưng em toàn test trên url thôi í chị, em test trên url thì nó chạy tốt cả =))
nhưng nãy em sửa lại dưới view dùng ajax thì lại không được
em có thêm đoạn render(:js => "window.location = '#{carts_path}'") if request.xhr? trong load_product chị xem ổn không nha

Reply:
nhưng có vấn đề là cái flash nó lại không có show lên được khi render í chị, em làm cái action hơi tởm chị xem giúp em với =)))